src/clj/ataru/haku/haku_service.clj

```diff
@@ -317,3 +318,13 @@
            :haut             haut
            :hakukohteet      hakukohteet-with-kevyt-valinta
            :hakukohderyhmat  hakukohderyhmat}))
+
+(defn get-haut-for-kk-application-payments
+  "Get hakus according to study start term and year. Only for internal use, does not do authorization."
+  [get-haut-cache tarjonta-service start-term start-year]
+  (->> (cache/get-from get-haut-cache :haut)
+       (map :haku)
+       distinct
+       (keep #(tarjonta/get-haku tarjonta-service %))
+       (filter #(and (= start-year (:alkamisvuosi %))
+                     (str/starts-with? (:alkamiskausi %) start-term)))))
```

Comment:
Jos näitä käytetään vain kk-hakuja varten sitten kannattaisi myös katsoa et haku on korkeakouluhaku

valintojen-toteuttamisessa se päätellään näin:

```
export function isKorkeakouluHaku(haku: Haku): boolean {
  return haku.kohdejoukkoKoodiUri.startsWith('haunkohdejoukko_12');
}
```

Reply:
Tämä tsekataan tuolla toisaalla pidemmällä putkessa (requires-higher-education-application-fee?), mutta voisi harkita että siirtäisi suoraan tänne jotta tulisi jo samantien fiksusti suodatettu joukko hakuja.